sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -2101,24 +2148,57 @@ private static CosmosException getNegativeTimeoutException(RxDocumentServiceRequ
     @Override
     public Mono<ResourceResponse<Document>> upsertDocument(String collectionLink, Object document,
                                                                  RequestOptions options, boolean disableAutomaticIdGeneration) {
-        DocumentClientRetryPolicy requestRetryPolicy = this.resetSessionTokenRetryPolicy.getRequestPolicy();
+        return wrapPointOperationWithAvailabilityStrategy(
```

Comment:
feel `getRxDocumentServiceResponseMonoWithE2ETimeout` logic should be moved to here after the change

Reply:
I will take a look